services/src/main/java/org/keycloak/broker/oidc/mappers/AbstractJsonUserAttributeMapper.java

```diff
@@ -206,9 +207,15 @@ public static String getJsonValue(JsonNode baseNode, String fieldPath) {
 			}
 
 			if (idx < 0) {
-				if (!currentNode.isValueNode()) {
+				if (currentNode.isArray()) {
```

Comment:
This changes semantics - warning has to be logged and null has to be returned when the current node is neither simple array of strings node nor value node